rllib/utils/replay_buffers/replay_buffer.py

```diff
@@ -80,15 +96,39 @@ def add(self, batch: SampleBatchType, **kwargs) -> None:
         assert batch.count > 0, batch
         warn_replay_capacity(item=batch, num_items=self.capacity / batch.count)
 
+        if self._storage_unit == StorageUnit.TIMESTEPS:
+            self._add_single_batch(batch, **kwargs)
+        elif self._storage_unit == StorageUnit.SEQUENCES:
+            timestep_count = 0
+            for seq_len in batch.get(SampleBatch.SEQ_LENS):
+                start_seq = timestep_count
+                end_seq = timestep_count + seq_len
+                self._add_single_batch(batch[start_seq:end_seq], **kwargs)
+                timestep_count = end_seq
+        elif self._storage_unit == StorageUnit.EPISODES:
+            for eps in batch.split_by_episode():
+                if eps.get(SampleBatch.T)[0] == 0 and \
```

Comment:
is this a new logic or copied from somewhere else?

Reply:
This is new logic and has not been tested with an RL algorithm.